test/include/dlaf_test/matrix/util_tile_blas.h

```diff
@@ -22,38 +24,46 @@ namespace dlaf {
 namespace matrix {
 namespace test {
 
-/// Sets the elements of the tile.
-///
-/// The (i, j)-element of the tile is set to el({i, j}) if op == NoTrans,
-///                                          el({j, i}) if op == Trans,
-///                                          conj(el({j, i})) if op == ConjTrans.
-/// @pre el argument is an index of type const TileElementIndex& or TileElementIndex,
-/// @pre el return type should be T.
-template <class T, class Func>
-void set(Tile<T, Device::CPU>& tile, Func el, blas::Op op) {
+namespace internal {
+template <class ElementGetter>
+auto opValFunc(ElementGetter val, const blas::Op op) {
+  std::function<decltype(val(std::declval<TileElementIndex>()))(TileElementIndex)> op_val;
   switch (op) {
     case blas::Op::NoTrans:
-      set(tile, el);
+      op_val = [&val](TileElementIndex i) { return val(i); };
       break;
 
     case blas::Op::Trans: {
-      auto op_el = [&el](TileElementIndex i) {
+      op_val = [&val](TileElementIndex i) {
         i.transpose();
-        return el(i);
+        return val(i);
       };
-      set(tile, op_el);
       break;
     }
 
     case blas::Op::ConjTrans: {
-      auto op_el = [&el](TileElementIndex i) {
+      op_val = [&val](TileElementIndex i) {
         i.transpose();
-        return dlaf::test::TypeUtilities<T>::conj(el(i));
+        return dlaf::conj(val(i));
       };
-      set(tile, op_el);
       break;
     }
   }
+  return op_val;
+}
+}
+
+/// Sets the elements of the tile.
+///
+/// The (i, j)-element of the tile is set to el({i, j}) if op == NoTrans,
```

Comment:
With (i,j) in the upper triangular part of the matrix?

I would take the chance to clarify in the documentation which part of the matrix acts as "source" for values.